test/src/helpers.h

```diff
@@ -42,6 +42,12 @@
 #include <string>
 #include <thread>
 
+namespace tiledb {
+namespace sm {
+class SubarrayPartitioner;
```

Comment:
Maybe simplify to `class tiledb::sm::SubarrayPartitioner;` ?

Reply:
IMHO, it cannot be done because **tiledb** and subnamespace **sm** will not be known by compiler yet. But another solution came to my mind, only **sm** namespace can be declared under namespace **tiledb** like:

```
namespace tiledb {

namespace sm {
class SubarrayPartitioner;
}

namespace test {
...
```